### ISSUE TITLE
Update from craft 3.4 to 3.5.3 breaks @formatDateTime directive

### ISSUE THREAD

**Issue by martijnvanonz** [craftcms/cms#6614]
### Description
Hi, wile updating from craft 3.4 to craft 3.5.3 all my Graphql querys broke on the @formatDateTime directive.
I'm getting an error: DateTimeZone::__construct() expects parameter 1 to be string, object given

### Steps to reproduce

1. My Graphql query:
```

export const SLIDER_CARD = gql`
  fragment event on event_event_Entry {
    ... on event_event_Entry {
      id
      title
      uri
      slug
      startDateTime @formatDateTime(format: "F j", timezone: "Europe/Amsterdam")
      endDateTime @formatDateTime(format: "F j", timezone: "Europe/Amsterdam")
      image {
        ${IMAGES('slider')}
      }
      venue {
        ... on venue_venue_Entry {
          id
          title
          postalCode
          city {
            ... on city_city_Entry {
              id
              title
            }
          }
        }
      }
    }
  }
`;

```
The response:
```

{
	"errors": [{
		"debugMessage": "DateTimeZone::__construct() expects parameter 1 to be string, object given",
		"message": "Internal server error",
		"category": "internal",
		"locations": [{
			"line": 212,
			"column": 5
		}],
		"path": ["events", 0, "startDateTime"]
	}, {
		"debugMessage": "DateTimeZone::__construct() expects parameter 1 to be string, object given",
		"message": "Internal server error",
		"category": "internal",
		"locations": [{
			"line": 212,
			"column": 5
		}],
		"path": ["events", 1, "startDateTime"]
	}, {
		"debugMessage": "DateTimeZone::__construct() expects parameter 1 to be string, object given",
		"message": "Internal server error",
		"category": "internal",
		"locations": [{
			"line": 212,
			"column": 5
		}],
		"path": ["events", 2, "startDateTime"]
	}, {
		"debugMessage": "DateTimeZone::__construct() expects parameter 1 to be string, object given",
		"message": "Internal server error",
		"category": "internal",
		"locations": [{
			"line": 212,
			"column": 5
		}],
		"path": ["events", 3, "startDateTime"]
	}, {
		"debugMessage": "DateTimeZone::__construct() expects parameter 1 to be string, object given",
		"message": "Internal server error",
		"category": "internal",
		"locations": [{
			"line": 212,
			"column": 5
		}],
		"path": ["events", 4, "startDateTime"]
	}, {
		"debugMessage": "DateTimeZone::__construct() expects parameter 1 to be string, object given",
		"message": "Internal server error",
		"category": "internal",
		"locations": [{
			"line": 212,
			"column": 5
		}],
		"path": ["events", 5, "startDateTime"]
	}],
	"data": {
		"entry": {
			"title": "Melting Ice:  An uncertain future",
			"seoTitle": null,
			"seoDescription": "SEO Description",
			"image": [{
				"placeholder": "http://localhost:8080/uploads/images/event/_120x80_crop_center-center_none/Event.png",
				"mobile": "http://localhost:8080/uploads/images/event/_480x322_crop_center-center_none/Event.png",
				"mobile2x": "http://localhost:8080/uploads/images/event/_960x644_crop_center-center_none/Event.png",
				"tablet": "http://localhost:8080/uploads/images/event/_1024x473_crop_center-center_none/Event.png",
				"tablet2x": "http://localhost:8080/uploads/images/event/_2048x946_crop_center-center_none/Event.png",
				"desktop": "http://localhost:8080/uploads/images/event/_1320x610_crop_center-center_none/Event.png",
				"desktop2x": "http://localhost:8080/uploads/images/event/_2640x1220_crop_center-center_none/Event.png",
				"title": "Event",
				"altText": null,
				"url": "http://localhost:8080/uploads/images/event/Event.png",
				"__typename": "images_Asset"
			}],
			"text": "<p>Science &amp; Cocktails is back with another home edition! We'll provide the science &amp; music, live streamed into your lazy chair. All you have to do is mix your own cocktails and tune into our Youtube channel. Fortunately, cocktail recipe and mixing tutorial will be provided.<br /><br />How is climate affecting ice sheets and glaciers and how do they affect the climate? What makes these ice sheets unstable? Can and will they reach tipping points and over what time scales is this possible? Can sea level rise by several meters by the end of the Century?<br /><br />All people on Earth depend directly or indirectly on ice. Around 10% of Earth’s land area is covered by glaciers or ice sheets and around 12% of the oceans is covered by floating ice. More and more evidence emerges every year showing that glaciers around the world are shrinking, and the Greenland and Antarctic ice sheets are getting smaller. <br /><br />But the holy grail questions concern their future: How much smaller will they get? How fast will they shrink? Moreover, if predicting how the climate will evolve is fiendishly difficult, how can we predict what ice sheets and glaciers will do? <br /><br />To answer these questions, we develop computer simulations of ice sheet and glacier behaviour and compare these models to real observations. But only recently have satellites been launched into space that can precisely weigh ice sheets and glaciers from afar, giving a concrete sense of how much ice is sloughing off. <br /><br />Today more than 200 million people in the world live in low lying coastal areas that also host many of the world's megacities and significant infrastructures. Rising sea levels due to melting ice poses a serious threat to coastal societies through an increased risk of flooding. Future sea level turns critically on the evolution of the large ice sheets in Antarctica and Greenland, together potentially accounting for more than 65 m of global sea-level rise. <br /><br />After the talk and a short break so you can crack some ice in your cocktail shaker and refill those glasses, we'll switch over to Macgray's house for a journey through his analog synths, classical piano and techno beats. He'll be joined by the majestic melodies of Hélène Van Loo on the flute.</p>",
			"facebookEventUrl": "https://www.facebook.com/events/576514949964414/",
			"contactEmail": "bxl@scienceandcocktails.org",
			"ticketsUrl": "https://www.atelier210.be/",
			"venue": [{
				"id": "149",
				"title": "Schouwburg",
				"address": "Schouwburgstraat 14",
				"postalCode": "1075SH",
				"city": [{
					"id": "147",
					"title": "Amsterdam",
					"__typename": "city_city_Entry"
				}],
				"__typename": "venue_venue_Entry"
			}],
			"startDate": "2020-08-26T18:30:00+00:00",
			"startTime": "2020-08-26T18:30:00+00:00",
			"endDateTime": "2020-08-26T21:00:00+00:00",
			"doorsOpen": "2020-08-12T17:30:00+00:00",
			"blocks": [{
				"id": "549",
				"typeHandle": "person",
				"preTitle": "Prof. Dr.",
				"imagePosition": "left",
				"person": [{
					"id": "87",
					"image": [{
						"placeholder": "http://localhost:8080/uploads/images/person/_120x120_crop_center-center_none/Person.png",
						"mobile": "http://localhost:8080/uploads/images/person/_480x480_crop_center-center_none/Person.png",
						"mobile2x": "http://localhost:8080/uploads/images/person/_960x960_crop_center-center_none/Person.png",
						"tablet": "http://localhost:8080/uploads/images/person/_600x600_crop_center-center_none/Person.png",
						"tablet2x": "http://localhost:8080/uploads/images/person/_1200x1200_crop_center-center_none/Person.png",
						"desktop": "http://localhost:8080/uploads/images/person/_1280x1280_crop_center-center_none/Person.png",
						"desktop2x": "http://localhost:8080/uploads/images/person/_2560x2560_crop_center-center_none/Person.png",
						"title": "Person",
						"altText": "Sergio",
						"url": "http://localhost:8080/uploads/images/person/Person.png",
						"__typename": "images_Asset"
					}],
					"title": "Sergio",
					"__typename": "person_person_Entry"
				}],
				"text": "<p>Professor at the Université libre de Bruxelles. He is best known for developing ice-sheet models and leading model intercomparisons. He took part in several expeditions to Antarctica and to glaciers of the Arctic in order to study the interaction of glaciers and ice sheets with subglacial water and the ocean. Pattyn has received the 2018 Louis Agassiz Medal, an award from the European Geosciences Union that recognises his outstanding scientific contribution to the study of the cryosphere.</p>",
				"linkTitle": "website",
				"linkUrl": "http://homepages.ulb.ac.be/~fpattyn/welcome.html",
				"internalPage": [],
				"__typename": "blocks_person_BlockType"
			}, {
				"id": "552",
				"typeHandle": "person",
				"preTitle": "music by",
				"imagePosition": "right",
				"person": [{
					"id": "51",
					"image": [{
						"placeholder": "http://localhost:8080/uploads/images/person/_120x120_crop_center-center_none/Person.png",
						"mobile": "http://localhost:8080/uploads/images/person/_480x480_crop_center-center_none/Person.png",
						"mobile2x": "http://localhost:8080/uploads/images/person/_960x960_crop_center-center_none/Person.png",
						"tablet": "http://localhost:8080/uploads/images/person/_600x600_crop_center-center_none/Person.png",
						"tablet2x": "http://localhost:8080/uploads/images/person/_1200x1200_crop_center-center_none/Person.png",
						"desktop": "http://localhost:8080/uploads/images/person/_1280x1280_crop_center-center_none/Person.png",
						"desktop2x": "http://localhost:8080/uploads/images/person/_2560x2560_crop_center-center_none/Person.png",
						"title": "Person",
						"altText": "Sergio",
						"url": "http://localhost:8080/uploads/images/person/Person.png",
						"__typename": "images_Asset"
					}],
					"title": "Inan Smith",
					"__typename": "person_person_Entry"
				}],
				"text": "<p>Influenced by neoclassical, ambient and techno music, Macgray's music is full of analog synths and piano, big kicks and subbasses, music for movies and electronica. The songs tell a story. The melodies slowly rise. The rhythms explode. For this special session, Macgray will be joined by the majestic melodies of the eclectic flutist, Hélène Van Loo. Hélène has composed music for films and currently performs with her Sarrazine project. She collaborates with a wide range of artists both on stage and in the studio and hosts a radio program on Musiq3</p>",
				"linkTitle": "Macgray video",
				"linkUrl": "https://www.youtube.com/watch?v=GH_OEoI_jTE",
				"internalPage": [],
				"__typename": "blocks_person_BlockType"
			}, {
				"id": "559",
				"typeHandle": "video",
				"mainTitle": "The talk",
				"videoPreTitle": "S&C@Home",
				"videoMainTitle": "\"Melting Ice\" with Frank Pattyn",
				"videoText": "<p>How is climate affecting ice sheets and glaciers and how do they affect the climate? What makes these ice sheets unstable? Can and will they reach tipping points and over what time scales is this possible? Can sea level rise by several meters by the end of the Century?</p>",
				"youtubeVideo": "youtube_sbmG1XA5TZg",
				"placeholderImage": [{
					"placeholder": "http://localhost:8080/uploads/images/blocks/_85x57_crop_center-center_none/SS_Founding-100_web-Charelle.jpg",
					"mobile": "http://localhost:8080/uploads/images/blocks/_343x231_crop_center-center_none/SS_Founding-100_web-Charelle.jpg",
					"mobile2x": "http://localhost:8080/uploads/images/blocks/_686x462_crop_center-center_none/SS_Founding-100_web-Charelle.jpg",
					"tablet": "http://localhost:8080/uploads/images/blocks/_600x338_crop_center-center_none/SS_Founding-100_web-Charelle.jpg",
					"tablet2x": "http://localhost:8080/uploads/images/blocks/_1200x676_crop_center-center_none/SS_Founding-100_web-Charelle.jpg",
					"desktop": "http://localhost:8080/uploads/images/blocks/_740x417_crop_center-center_none/SS_Founding-100_web-Charelle.jpg",
					"desktop2x": "http://localhost:8080/uploads/images/blocks/_1480x834_crop_center-center_none/SS_Founding-100_web-Charelle.jpg",
					"title": "SS Founding 100 web Charelle",
					"altText": "Sergio33",
					"url": "http://localhost:8080/uploads/images/blocks/SS_Founding-100_web-Charelle.jpg",
					"__typename": "images_Asset"
				}],
				"__typename": "blocks_video_BlockType"
			}],
			"__typename": "event_event_Entry"
		},
		"globalSet": {
			"defaultTitleSuffix": "Default Title",
			"defaultMetaDescription": "Default meta",
			"defaultShareImage": [{
				"url": "http://localhost:8080/uploads/images/Rectangle-142.png",
				"__typename": "images_Asset"
			}],
			"__typename": "seoSettings_GlobalSet"
		},
		"events": [{
			"id": "213",
			"title": "Danswedstrijd Breda copy",
			"uri": "event/johannesburg/danswedstrijd-breda-1",
			"slug": "danswedstrijd-breda-1",
			"startDateTime": null,
			"endDateTime": "2020-09-22T00:00:00+00:00",
			"image": [{
				"placeholder": "http://localhost:8080/uploads/images/event/_53x50_crop_center-center_none/Event.png",
				"mobile": "http://localhost:8080/uploads/images/event/_215x203_crop_center-center_none/Event.png",
				"mobile2x": "http://localhost:8080/uploads/images/event/_430x406_crop_center-center_none/Event.png",
				"tablet": "http://localhost:8080/uploads/images/event/_341x317_crop_center-center_none/Event.png",
				"tablet2x": "http://localhost:8080/uploads/images/event/_682x634_crop_center-center_none/Event.png",
				"desktop": "http://localhost:8080/uploads/images/event/_600x558_crop_center-center_none/Event.png",
				"desktop2x": "http://localhost:8080/uploads/images/event/_1200x1116_crop_center-center_none/Event.png",
				"title": "Event",
				"altText": null,
				"url": "http://localhost:8080/uploads/images/event/Event.png",
				"__typename": "images_Asset"
			}],
			"venue": [{
				"id": "141",
				"title": "Marktplein",
				"postalCode": "1075SH",
				"city": [{
					"id": "366",
					"title": "Johannesburg",
					"__typename": "city_city_Entry"
				}],
				"__typename": "venue_venue_Entry"
			}],
			"__typename": "event_event_Entry"
		}, {
			"id": "143",
			"title": "Danswedstrijd Breda",
			"uri": "event/johannesburg/danswedstrijd-breda",
			"slug": "danswedstrijd-breda",
			"startDateTime": null,
			"endDateTime": "2020-09-20T00:00:00+00:00",
			"image": [{
				"placeholder": "http://localhost:8080/uploads/images/event/_53x50_crop_center-center_none/Event.png",
				"mobile": "http://localhost:8080/uploads/images/event/_215x203_crop_center-center_none/Event.png",
				"mobile2x": "http://localhost:8080/uploads/images/event/_430x406_crop_center-center_none/Event.png",
				"tablet": "http://localhost:8080/uploads/images/event/_341x317_crop_center-center_none/Event.png",
				"tablet2x": "http://localhost:8080/uploads/images/event/_682x634_crop_center-center_none/Event.png",
				"desktop": "http://localhost:8080/uploads/images/event/_600x558_crop_center-center_none/Event.png",
				"desktop2x": "http://localhost:8080/uploads/images/event/_1200x1116_crop_center-center_none/Event.png",
				"title": "Event",
				"altText": null,
				"url": "http://localhost:8080/uploads/images/event/Event.png",
				"__typename": "images_Asset"
			}],
			"venue": [{
				"id": "141",
				"title": "Marktplein",
				"postalCode": "1075SH",
				"city": [{
					"id": "366",
					"title": "Johannesburg",
					"__typename": "city_city_Entry"
				}],
				"__typename": "venue_venue_Entry"
			}],
			"__typename": "event_event_Entry"
		}, {
			"id": "215",
			"title": "Franco 9.45pm",
			"uri": "event/amsterdam/franco",
			"slug": "franco",
			"startDateTime": null,
			"endDateTime": "2020-09-19T21:00:00+00:00",
			"image": [{
				"placeholder": "http://localhost:8080/uploads/images/event/_53x50_crop_center-center_none/Event.png",
				"mobile": "http://localhost:8080/uploads/images/event/_215x203_crop_center-center_none/Event.png",
				"mobile2x": "http://localhost:8080/uploads/images/event/_430x406_crop_center-center_none/Event.png",
				"tablet": "http://localhost:8080/uploads/images/event/_341x317_crop_center-center_none/Event.png",
				"tablet2x": "http://localhost:8080/uploads/images/event/_682x634_crop_center-center_none/Event.png",
				"desktop": "http://localhost:8080/uploads/images/event/_600x558_crop_center-center_none/Event.png",
				"desktop2x": "http://localhost:8080/uploads/images/event/_1200x1116_crop_center-center_none/Event.png",
				"title": "Event",
				"altText": null,
				"url": "http://localhost:8080/uploads/images/event/Event.png",
				"__typename": "images_Asset"
			}],
			"venue": [{
				"id": "149",
				"title": "Schouwburg",
				"postalCode": "1075SH",
				"city": [{
					"id": "147",
					"title": "Amsterdam",
					"__typename": "city_city_Entry"
				}],
				"__typename": "venue_venue_Entry"
			}],
			"__typename": "event_event_Entry"
		}, {
			"id": "151",
			"title": "Vishengel tentoonstelling Roermond",
			"uri": "event/amsterdam/vishengel-tentoonstelling-roermond",
			"slug": "vishengel-tentoonstelling-roermond",
			"startDateTime": null,
			"endDateTime": "2020-09-18T00:00:00+00:00",
			"image": [{
				"placeholder": "http://localhost:8080/uploads/images/event/_53x50_crop_center-center_none/Event.png",
				"mobile": "http://localhost:8080/uploads/images/event/_215x203_crop_center-center_none/Event.png",
				"mobile2x": "http://localhost:8080/uploads/images/event/_430x406_crop_center-center_none/Event.png",
				"tablet": "http://localhost:8080/uploads/images/event/_341x317_crop_center-center_none/Event.png",
				"tablet2x": "http://localhost:8080/uploads/images/event/_682x634_crop_center-center_none/Event.png",
				"desktop": "http://localhost:8080/uploads/images/event/_600x558_crop_center-center_none/Event.png",
				"desktop2x": "http://localhost:8080/uploads/images/event/_1200x1116_crop_center-center_none/Event.png",
				"title": "Event",
				"altText": null,
				"url": "http://localhost:8080/uploads/images/event/Event.png",
				"__typename": "images_Asset"
			}],
			"venue": [{
				"id": "149",
				"title": "Schouwburg",
				"postalCode": "1075SH",
				"city": [{
					"id": "147",
					"title": "Amsterdam",
					"__typename": "city_city_Entry"
				}],
				"__typename": "venue_venue_Entry"
			}],
			"__typename": "event_event_Entry"
		}, {
			"id": "155",
			"title": "Sergio 7pm",
			"uri": "event/amsterdam/sergio",
			"slug": "sergio",
			"startDateTime": null,
			"endDateTime": "2020-09-17T18:15:00+00:00",
			"image": [{
				"placeholder": "http://localhost:8080/uploads/images/event/_53x50_crop_center-center_none/Event.png",
				"mobile": "http://localhost:8080/uploads/images/event/_215x203_crop_center-center_none/Event.png",
				"mobile2x": "http://localhost:8080/uploads/images/event/_430x406_crop_center-center_none/Event.png",
				"tablet": "http://localhost:8080/uploads/images/event/_341x317_crop_center-center_none/Event.png",
				"tablet2x": "http://localhost:8080/uploads/images/event/_682x634_crop_center-center_none/Event.png",
				"desktop": "http://localhost:8080/uploads/images/event/_600x558_crop_center-center_none/Event.png",
				"desktop2x": "http://localhost:8080/uploads/images/event/_1200x1116_crop_center-center_none/Event.png",
				"title": "Event",
				"altText": null,
				"url": "http://localhost:8080/uploads/images/event/Event.png",
				"__typename": "images_Asset"
			}],
			"venue": [{
				"id": "149",
				"title": "Schouwburg",
				"postalCode": "1075SH",
				"city": [{
					"id": "147",
					"title": "Amsterdam",
					"__typename": "city_city_Entry"
				}],
				"__typename": "venue_venue_Entry"
			}],
			"__typename": "event_event_Entry"
		}, {
			"id": "546",
			"title": "Melting Ice:  An uncertain future",
			"uri": "event/amsterdam/melting-ice-an-uncertain-future",
			"slug": "melting-ice-an-uncertain-future",
			"startDateTime": null,
			"endDateTime": "2020-08-26T21:00:00+00:00",
			"image": [{
				"placeholder": "http://localhost:8080/uploads/images/event/_53x50_crop_center-center_none/Event.png",
				"mobile": "http://localhost:8080/uploads/images/event/_215x203_crop_center-center_none/Event.png",
				"mobile2x": "http://localhost:8080/uploads/images/event/_430x406_crop_center-center_none/Event.png",
				"tablet": "http://localhost:8080/uploads/images/event/_341x317_crop_center-center_none/Event.png",
				"tablet2x": "http://localhost:8080/uploads/images/event/_682x634_crop_center-center_none/Event.png",
				"desktop": "http://localhost:8080/uploads/images/event/_600x558_crop_center-center_none/Event.png",
				"desktop2x": "http://localhost:8080/uploads/images/event/_1200x1116_crop_center-center_none/Event.png",
				"title": "Event",
				"altText": null,
				"url": "http://localhost:8080/uploads/images/event/Event.png",
				"__typename": "images_Asset"
			}],
			"venue": [{
				"id": "149",
				"title": "Schouwburg",
				"postalCode": "1075SH",
				"city": [{
					"id": "147",
					"title": "Amsterdam",
					"__typename": "city_city_Entry"
				}],
				"__typename": "venue_venue_Entry"
			}],
			"__typename": "event_event_Entry"
		}]
	}
}

```

### Additional info

- Craft version: 3.5.3
- PHP version: 7.4

**Comment by brandonkelly** [comment]
I’m not able to reproduce this. Are you able to find that error in your logs (`storage/logs/`)? If so can you please post the stack trace that follows?

**Comment by martijnvanonz** [comment]
Hi @brandonkelly,

I can not find the error back in the logs. But after some investigation I found this:

In FormatDateTime.php in the apply function.

A new DateTimeZone object is made, on line 87
```
$timezone = new \DateTimeZone($arguments['timezone'] ?? self::DEFAULT_TIMEZONE);
```

And is used in the formatter, on line 105.

```
$formatter->timeZone = $timezone;
```

When i check the yii Formatter.php I see that the function expects a string.

```
 /**
     * @var string the time zone to use for formatting time and date values.
     *
     * This can be any value that may be passed to [date_default_timezone_set()](https://secure.php.net/manual/en/function.date-default-timezone-set.php)
     * e.g. `UTC`, `Europe/Berlin` or `America/Chicago`.
     * Refer to the [php manual](https://secure.php.net/manual/en/timezones.php) for available time zones.
     * If this property is not set, [[\yii\base\Application::timeZone]] will be used.
     *
     * Note that the default time zone for input data is assumed to be UTC by default if no time zone is included in the input date value.
     * If you store your data in a different time zone in the database, you have to adjust [[defaultTimeZone]] accordingly.
     */
    public $timeZone;
```
Changing 
```
$timezone = new \DateTimeZone($arguments['timezone'] ?? self::DEFAULT_TIMEZONE);
```

to 

```
$timezone = $arguments['timezone'] ?? self::DEFAULT_TIMEZONE;
```

Seems to fix it but I don't know if it is the right solution.

Thank you for your time!

**Comment by andris-sevcenko** [comment]
Can you check whether you have the `Intl` PHP extension enabled?

**Comment by martijnvanonz** [comment]
Hi Andris.

I checked the Intl extension is not installed. I will install it and check it again.

**Comment by martijnvanonz** [comment]
Hi Both,

Installing the Intl extension fixed the problem. My bad!

Thank you both for your help.

**Comment by andris-sevcenko** [comment]
@martijnvanonz it's an optional extension, so this is still an issue to be fixed :)

**Comment by andris-sevcenko** [comment]
All fixed for the next 3.5 release!

**Comment by brandonkelly** [comment]
Craft 3.5.4 is out now with the fix for this.
